pkg/config/config.go

```diff
@@ -0,0 +1,175 @@
+package config
+
+import (
+	"errors"
+	"github.com/BurntSushi/toml"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type (
+	// Config provides a general structure to capture the config options
+	// for the toolchain. This may be abstracted out to a common library in
+	// the future as more components are added to the toolchain.
+	Config struct {
+		Logger  Logger  `toml:"logger"`
+		Scanner Scanner `toml:"scanner"`
+	}
+
+	// Logger provides general logging config
+	Logger struct {
+		Level string `toml:"level"`
+	}
+
+	// Scanner provides scanner specific config
+	Scanner struct {
+		Workdir  string   `toml:"workdir"`
+		Gitleaks Gitleaks `toml:"gitleaks"`
+		Patterns Patterns `toml:"patterns"`
+	}
+
+	// Gitleaks configures the gitleaks subscanner
+	Gitleaks struct {
+		Version string `toml:"version"`
+	}
+
+	// Patterns provides configuration for managing pattern updates
+	Patterns struct {
+		RefreshInterval int           `toml:"refresh_interval"`
+		Server          PatternServer `toml:"server"`
+	}
+
+	// PatternServer provides pattern server configuration settings for the scanner
+	PatternServer struct {
+		URL       string `toml:"url"`
+		AuthToken string `toml:"auth_token"`
+	}
+)
+
+func xdgDir(defaultName, envVar string) string {
```

Comment:
Would you consider using something like https://github.com/adrg/xdg to do this cross platform? This ones dependencies are already in here anyway.